### PR TITLE
Refactoring makefile.

### PR DESCRIPTION
**Pull request by arhamchordia**
## 1. Overview

Restructuring makefile to keep make commands more readable and easy to execute.

## 2. Implementation details

Adding different makefiles in scripts/makefiles
Separating them into : 
1. build.mk
2. docker.mk 
3. and so on....

## 3. How to test/use

Run commands one by one to test them.

### CODE REVIEW

File: scripts/makefiles/proto.mk [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	@echo "Available Commands:"
+	@echo "  all        Run proto-format and proto-gen"
+	@echo "  gen        Generate Protobuf files"
+	@echo "  gen-1      Generate Protobuf files (old relic)"
+	@echo "  doc        Generate proto docs"
+
+proto: proto-help
+proto-all: proto-gen
+
+# todo : @AJ needs to address this after removing third_party. Refer this for removal https://github.com/osmosis-labs/osmosis/blob/188abfcd15544ca07d468c0dc0169876ffde6079/scripts/makefiles/proto.mk#L39
```
Comment (magiodev) [inline scripts/makefiles/proto.mk:19]: Is this TODO for now?

**Review by magiodev — COMMENTED** [review]
Refactor looks good. I see lot of redundancy over commands, should we try to reduce the amount of repeated or similar ones?

**Review by akure — COMMENTED** [review]
The structure is good. Is there any way to test all the Makefile commands in one go in local.  Like a command that will build, test and try maximum possibilities.  Not necessarily to be part of this PR.

File: scripts/makefiles/proto.mk (hunk 1 above)
Comment (arhamchordia) [reply scripts/makefiles/proto.mk:19]: Not for now. To be addressed by @ajansari95 in his cleanup PR

File: scripts/makefiles/proto.mk [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+proto-gen:
+	@echo "Generating Protobuf files"
+	@sh ./scripts/protocgen.sh
+
+# todo : @AK need the reason why it was there earlier
+proto-gen-1:
```
[the remark this answers is not in the document]
Comment (arhamchordia) [reply scripts/makefiles/proto.mk:25]: Sure. So we can get rid of it now?

**Comment by arhamchordia** [comment]
> The structure is good. Is there any way to test all the Makefile commands in one go in local. Like a command that will build, test and try maximum possibilities. Not necessarily to be part of this PR.

Yes. We can make a general command in makefile that includes all of them together. Something like `make execute-all` which in turn executes everything.

**Comment by arhamchordia** [comment]
> Refactor looks good. I see lot of redundancy over commands, should we try to reduce the amount of repeated or similar ones?

I see only distroless build as the redundant one. Any others that you could find?

**Review by akure — APPROVED** [review]
lg

**Pull request merged**